### PR TITLE
Add RegisterAgent rpcs, update libraries, update error handling

### PR DESCRIPTION
**Pull request by adjackura**
RegisterAgent currently unused
agentendpoint tests had to be updated because of new client library retries (really wasn't testing something that we cared about)
Moved a few error handling to new go 1.13 errors.Is and errors.As

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -83,6 +84,8 @@ var (
 	agentConfigMx sync.RWMutex
 	version       string
 	lEtag         = &lastEtag{Etag: "0"}
+
+	capabilities = []string{"PATCH_GA", "GUEST_POLICY_BETA"}
```
Comment (jaiminsh) [inline config/config.go:88]: Maybe have a comment on how these values are decided?

Comment (adjackura) [reply config/config.go:91]: Made a comment for how they are used, i don't really want to publish how we decide them beyond this.

**Pull request merged**